Detectors/ITSMFT/MFT/tracking/include/MFTTracking/TrackParam.h

```diff
@@ -0,0 +1,182 @@
+  void setCovariances(const Double_t matrix[5][5]);
+  void setVariances(const Double_t matrix[5][5]);
+  void deleteCovariances();
+
+  const TMatrixD& getPropagator() const;
+  void resetPropagator();
+  void updatePropagator(const TMatrixD& propagator);
+
+  const TMatrixD& getExtrapParameters() const;
+  void setExtrapParameters(const TMatrixD& parameters);
+
+  const TMatrixD& getExtrapCovariances() const;
+  void setExtrapCovariances(const TMatrixD& covariances);
+
+  const TMatrixD& getSmoothParameters() const;
+  void setSmoothParameters(const TMatrixD& parameters);
+
+  const TMatrixD& getSmoothCovariances() const;
+  void setSmoothCovariances(const TMatrixD& covariances);
+
+  /// get pointer to associated cluster
+  const o2::itsmft::Cluster* getClusterPtr() const { return mClusterPtr; }
+  /// set pointer to associated cluster
+  void setClusterPtr(const o2::itsmft::Cluster* cluster) { mClusterPtr = cluster; }
+
+  /// return true if the associated cluster can be removed from the track it belongs to
+  Bool_t isRemovable() const { return mRemovable; }
+  /// set the flag telling whether the associated cluster can be removed from the track it belongs to or not
+  void setRemovable(Bool_t removable) { mRemovable = removable; }
+
+  /// return the chi2 of the track when the associated cluster was attached
+  Double_t getTrackChi2() const { return mTrackChi2; }
+  /// set the chi2 of the track when the associated cluster was attached
+  void setTrackChi2(Double_t chi2) { mTrackChi2 = chi2; }
+  /// return the local chi2 of the associated cluster with respect to the track
+  Double_t getLocalChi2() const { return mLocalChi2; }
+  /// set the local chi2 of the associated cluster with respect to the track
+  void setLocalChi2(Double_t chi2) { mLocalChi2 = chi2; }
+
+  Bool_t isCompatibleTrackParam(const TrackParam& trackParam, Double_t sigma2Cut, Double_t& normChi2) const;
+
+  void print() const;
+
+  void clear();
+
+ private:
+  Double_t mZ = 0.; ///< Z coordinate (cm)
+
+  /// Track parameters ordered as follow:      <pre>
+  /// X       = X coordinate   (cm)
+  /// Y       = Y coordinate   (cm)
+  /// PHI     = azimutal angle
+  /// TANL    = tangent of \lambda (dip angle)
+  /// INVQPT    = Inverse transverse momentum (GeV/c ** -1) times charge (assumed forward motion)  </pre>
+  TMatrixD mParameters{5, 1}; ///< \brief Track parameters
+
+  /// Covariance matrix of track parameters, ordered as follows:    <pre>
+  ///  <X,X>         <Y,X>           <PHI,X>       <TANL,X>        <INVQPT,X>
+  ///  <X,Y>         <Y,Y>           <PHI,Y>       <TANL,Y>        <INVQPT,Y>
+  /// <X,PHI>       <Y,PHI>         <PHI,PHI>     <TANL,PHI>      <INVQPT,PHI>
+  /// <X,TANL>      <Y,TANL>       <PHI,TANL>     <TANL,TANL>     <INVQPT,TANL>
+  /// <X,INVQPT>   <Y,INVQPT>     <PHI,INVQPT>   <TANL,INVQPT>   <INVQPT,INVQPT>  </pre>
+  mutable std::unique_ptr<TMatrixD> mCovariances{}; ///< \brief Covariance matrix of track parameters
```

Comment:
The cov.matrix is symmetric, either use array of 15 values (preferable) or ``ROOT::Math::SMatrix<double, 5, 5, ROOT::Math::MatRepSym<double, 5>>;``, avoid TMatrixD at any rate!

Reply:
See comment bellow. SMatrix will most certainly be used.